### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by funkylen**
- Created user model
- Created user controller with index, show and destroy methods
- Created profile controller with me, auth, register and edit methods 

Available routes:
`GET /api/users`
`GET /api/users/{id}`
`DELETE /api/users/{id}`

`GET /api/profile`
`PUT /api/profile`
`POST /api/profile/auth`
`POST /api/profile/register`

### CODE REVIEW

**Comment by VitalyKrenel** [comment]
Do you mind rebasing your auth branch onto scaffold, so I can see the branch changes a bit more easily?

**Comment by funkylen** [comment]
How, if i push in empty master branch?

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  has_secure_password
+
+  validates :password, length: {in: 6..20}, presence: true
+  validates :email, uniqueness: true, format: {
+      with: /\A(|(([A-Za-z0-9]+_+)|([A-Za-z0-9]+\-+)|([A-Za-z0-9]+\.+)|([A-Za-z0-9]+\++))*[A-Za-z0-9]+@((\w+\-+)|(\w+\.))*\w{1,63}\.[a-zA-Z]{2,6})\z/i,
+      message: 'invalid email format'
+  }, allow_blank: true
+  validates :login, uniqueness: true, length: {minimum: 3}, allow_blank: true
+  validates :name, length: {minimum: 3}, allow_blank: true
```
Comment (VitalyKrenel) [inline app/models/user.rb]: Overall, I don't recommend really building strict validation rules. `Yi` is pretty valid name in my opinion, but this way it won't fit. 

It actually applies to all personal information - it is better to not validate at all and use raw data, then apply counterfeit validations when there are no clearly defined rules from analytics or product owners.

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,13 @@
 Rails.application.routes.draw do
+  scope 'api' do
+    resources :users
+
+    scope 'profile' do
+      get '/', to: 'profile#me'
+      put '/', to: 'profile#edit'
+      post 'auth', to: 'profile#auth'
```
Comment (VitalyKrenel) [inline config/routes.rb]: Good idea to have a `/profile` to get a currently logged user information. 👍 

Btw, just curious: why you decided to make registration and login processes part of the `/profile`? 

In my opinion, profile seems to be focused on user's data and editing it, but not on the authentication and user creation processes. I would probably go with `UserController` handling user registration (creation of a new user) and `AuthenticationController` (or SessionController as we imply user work session, even though we use JWT) handling `login` and `logout` actions.

Anyway, seems to me we have a working token based authentication process now to use later on client side!

Comment (funkylen) [reply config/routes.rb]: It is semantically understandable for me:
- get profile data
- edit profile
- register profile 
- login to profile

If it is not so obviously, i can move out auth and register from profile scope

**Pull request merged**